### DISCUSSION TITLE
Is it capable of using offline pre-downloaded osm files to render it?

### DISCUSSION THREAD

**Discussion by liagha** [podusowski/walkers#217] [General]

**Comment by podusowski** [comment]
If you mean the rendered tiles, then there is a support for [HTTP cache](https://docs.rs/walkers/latest/walkers/struct.HttpOptions.html#), but that's it. There is no explicit API to download them beforehand. Also, [it would be against osm policies](https://operations.osmfoundation.org/policies/tiles/).

**Comment by liagha** [comment]
In the openstreetmaps website theres a part where you select an area and from there you can download osm files with that info. Im trying to use those files in egui

**Comment by podusowski** [comment]
I see. I'm afraid this library cannot render these. You would need to use Mapnik or similar to generate tiles, and then you would display them in Walkers. It's a quite a big task though.
